### PR TITLE
[MGDOBR-1111] : Sonarcloud NPE Fix

### PR DESCRIPTION
**Pull request by Aakanksha0311**
[MGDOBR-1111](https://issues.redhat.com/browse/MGDOBR-1111)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java [hunk 1]
```diff
@@ -179,12 +179,16 @@ private Processor doCreateProcessor(Bridge bridge, String customerId, String own
     }
 
     private Action resolveAction(Action action, String customerId, String bridgeId, String processorId) {
+        if(action == null)
```
Comment (sutaakar) [inline manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java:182]: please add curly braces for `if` clauses

File: manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java [hunk 2]
```diff
@@ -179,12 +179,16 @@ private Processor doCreateProcessor(Bridge bridge, String customerId, String own
...
+            throw new ItemNotFoundException("Action is null");
```
Comment (sutaakar) [inline manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java:183]: I think that in this case you can return `null`

File: manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java (hunk 1 above)
Comment (Aakanksha0311) [reply manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java]: Ok. Added.

File: manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java [hunk 3]
```diff
@@ -23,19 +17,11 @@
 import com.redhat.service.smartevents.infra.exceptions.BridgeErrorHelper;
 import com.redhat.service.smartevents.infra.exceptions.BridgeErrorInstance;
 import com.redhat.service.smartevents.infra.exceptions.definitions.platform.InternalPlatformException;
-import com.redhat.service.smartevents.infra.exceptions.definitions.user.AlreadyExistingItemException;
-import com.redhat.service.smartevents.infra.exceptions.definitions.user.BadRequestException;
-import com.redhat.service.smartevents.infra.exceptions.definitions.user.BridgeLifecycleException;
-import com.redhat.service.smartevents.infra.exceptions.definitions.user.ItemNotFoundException;
-import com.redhat.service.smartevents.infra.exceptions.definitions.user.ProcessorLifecycleException;
+import com.redhat.service.smartevents.infra.exceptions.definitions.user.*;
```
Comment (sutaakar) [inline manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java:20]: Can you please revert the import changes?

Comment (Aakanksha0311) [reply manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java:20]: Sure, it's caused by the formatting it seems.

Comment (Aakanksha0311) [reply manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java]: Done.

**Review by jomarko — APPROVED** [review]
The sonar issue from main branch, reported fro `ProcessorServiceImpl` seems to be gone on this branch. Approving.

**Pull request merged**